soc/arm/st_stm32/common/soc_gpio.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2018 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * Derived from scattered soc_gpio.c files under soc/st_stm32/stm32f*.
+ * Originally written by:
+ * Open-RnD Sp. z o.o. (stm32l4/soc_gpio.c, stm32f1/soc_gpio.c)
+ * BayLibre, SAS (stm32l4/soc_gpio.c)
+ * Endre Karlson <endre.karlson@gmail.com> (stm32l0/soc_gpio.c)
+ * Yurii Hamann (stm32f7/soc_gpio.c)
+ * Linaro Limited (stm32f4/soc_gpio.c)
+ * RnDity Sp. z o.o. (stm32f3/soc_gpio.c, stm32f0/soc_gpio.c)
+ * qianfan Zhao <qianfanguijin@163.com> (stm32f2/soc_gpio.c)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief
+ *
+ * Common manipulation functions for GPIO of stm32 series.
+ *
+ */
+
+#include <errno.h>
+
+#include <gpio.h>
+#include <gpio/gpio_stm32.h>
+#include <soc.h>
+#include "soc_syscfg_common.h"
+
+/* F1 series STM32 use AFIO register to configure interrupt generation
+ * and pinmux.
+ */
+#ifdef CONFIG_SOC_SERIES_STM32F1X
```

Comment:
Shouldn't this be removed in the LL commit?

Reply:
Yes, and should syscfg_registers.h be removed?? Since we prefer using the LL library so they may will not be used in the future.